src/Castle.Core.Tests/DynamicProxy.Tests/PersistentProxyBuilderTestCase.cs

```diff
@@ -44,6 +53,107 @@ public void PersistentProxyBuilder_SavesSignedFile()
 			Assert.IsTrue(path.EndsWith(ModuleScope.DEFAULT_FILE_NAME));
 		}
 	}
-}
 
-#endif
\ No newline at end of file
+#elif NET9_0_OR_GREATER
+
+	[TestFixture]
+	[FixtureLifeCycle(LifeCycle.InstancePerTestCase)]
+	public class PersistentProxyBuilderTestCase
+	{
+		// The above fixture life cycle gives each test its own fresh fixture, such that each test.
+		// starts with an empty `assemblyPaths` list. This is important for parallel test execution
+		// to work correctly.
+		private List<string> assemblyPaths = new();
+
+		[SetUp]
+		public void EnsureNoAssembliesSaved()
+		{
+			Assume.That(assemblyPaths, Is.Empty, "Test did not start with an empty assembly paths list.");
+		}
+
+		[TearDown]
+		public void DeleteSavedAssemblies()
+		{
+			foreach (var assemblyPath in assemblyPaths)
+			{
+				File.Delete(assemblyPath);
+			}
+		}
+
+		[Test]
+		public void SavesOneAssemblyPerProxiedType()
+		{
+
+			var builder = new PersistentProxyBuilder();
+			builder.AssemblySaved += assemblyPaths.Add;
```

Comment:
This (and the same code in the other test) could probably go into the `[SetUp]` method, too, since that one already has dealings with `assemblyPaths`.